src/helm/benchmark/static/benchmarking.js

```diff
@@ -1186,96 +1180,120 @@ $(function () {
-    }),
-  ).then(() => {
-    if (urlParams.models) {
-      // Models
-      $main.empty()
-      $main.append(renderHeader('Models', renderModels()));
-      refreshHashLocation();
-    } else if (urlParams.scenarios) {
-      // Models
-      $main.empty()
-      $main.append(renderHeader('Scenarios', renderScenarios()));
-      refreshHashLocation();
-    } else if (urlParams.plots) {
-      // Plots
-      $main.empty()
-      $main.append(renderHeader('Plots', renderPlots()));
-      refreshHashLocation();
-    } else if (urlParams.runSpec || urlParams.runSpecs || urlParams.runSpecRegex) {
-      // Predictions for a set of run specs (matching a regular expression)
-      $main.text('Loading runs...');
-      $.getJSON(runSpecsJsonUrl(suite), {}, (response) => {
-        $main.empty();
-        const runSpecs = response;
-        console.log('runSpecs', runSpecs);
-        let matcher;
-        if (urlParams.runSpec) {
-          // Exactly one
-          matcher = (runSpec) => runSpec.name === urlParams.runSpec;
-        } else if (urlParams.runSpecs) {
-          // List
-          const selectedRunSpecs = JSON.parse(urlParams.runSpecs);
-          matcher = (runSpec) => selectedRunSpecs.includes(runSpec.name);
-        } else if (urlParams.runSpecRegex) {
-          // Regular expression
-          const regex = new RegExp('^' + urlParams.runSpecRegex + '$');
-          matcher = (runSpec) => regex.test(runSpec.name);
+  var runsToRunSuites = null;
+  const urlParams = decodeUrlParams(window.location.search);
+  var usingRelease;
+
+  $.getJSON('using_release', {}, (response) => {
+    usingRelease = response['use_release'];
+  }).then(() => {
+    // Extract the name of the release/suite from the URL parameters. Default to "latest" if none is specified.
+    var version;
+    if (usingRelease) {
+      version = "release" in urlParams ? urlParams.release : "latest";
+    } else {
+      version = "suite" in urlParams ? urlParams.suite : "latest";
+    }
+    console.log(`Version: ${version}`);
+    $.when(
+      $.get('schema.yaml', {}, (response) => {
+        const raw = jsyaml.load(response);
+        console.log('schema', raw);
+        schema = new Schema(raw);
+      }),
+      $.get(summaryJsonUrl(version, usingRelease), {}, (response) => {
+        console.log('summary', response);
+        summary = response;
+        if (usingRelease) {
+          $summary.append(`Release ${summary.release} (last updated ${summary.date})`);
         } else {
-          throw 'Internal error';
+          $summary.append(`Suite ${summary.suite} (last updated ${summary.date})`);
         }
-        const matchedRunSpecs = runSpecs.filter(matcher);
-        if (matchedRunSpecs.length === 0) {
-          $main.append(renderError('No matching runs'));
+      }),
+    ).then(() => {
+      $.get(runsToRunSuitesJsonUrl(version, usingRelease), {}, (response) => {
+          runsToRunSuites = response;
+      }).always(() => {
```

Comment:
Since the run_to_run_suites.json file may NOT be present depending on if the user is using the `suite` functionality, we use `.always` here rather than `.then` since it will error out in the suite case.

Not sure if there's a better way to handle this, though?

Reply:
You want something like this:

```javascript
    console.log(`Version: ${version}`);

    const schemaPromise = $.get('schema.yaml', {}, (response) => {
      const raw = jsyaml.load(response);
      console.log('schema', raw);
      schema = new Schema(raw);
    });

    const summaryPromise = $.get(summaryJsonUrl(version, usingRelease), {}, (response) => {
      console.log('summary', response);
      summary = response;
      if (usingRelease) {
        $summary.append(`Release ${summary.release} (last updated ${summary.date})`);
      } else {
        $summary.append(`Suite ${summary.suite} (last updated ${summary.date})`);
      }
    });

    const getRunToRunSuitesPromise = usingRelease ?
      $.get(runsToRunSuitesJsonUrl(version, usingRelease), {}) :
      $.Deferred().resolve({});
    const runToRunSuitesPromise = getRunToRunSuitesPromise.then((response) => {
      runsToRunSuites = response;
    })

    $.when(schemaPromise, summaryPromise, runToRunSuitesPromise).then(() => {
      if (urlParams.models) {
        // Models
        $main.empty()
        $main.append(renderHeader('Models', renderModels()));
        refreshHashLocation();
      } else if (urlParams.scenarios) {
        //...
```

Make sure to also modify the closing braces / brackets after doing this.

Reply:
Ahhhh yes, that's MUCH better, thank you!